File: ds/queue/myqueue.py
class myQueue:
    def __init__(self):
        self.items = []

    def enqueue(self, item):
        self.items.append(item)

    def dequeue(self):
        if self.is_empty():
            return None
        item = self.items[0]
        del self.items[0]
        return item

    def is_empty(self):
        return len(self.items) == 0

q = myQueue()
q.enqueue('a')
q.enqueue('b')
q.enqueue('c')

print(q.dequeue())  # a
print(q.dequeue())  # b
print(q.dequeue())  # c

File: ds/queue/myqueue.cpp
#include <iostream>
#include <vector>
using namespace std;

class myQueue {
private:
    vector<string> items;

public:
    void enqueue(string item) {
        items.push_back(item);
    }

    string dequeue() {
        if (is_empty()) {
            return "";
        }
        string item = items.front();
        items.erase(items.begin());
        return item;
    }

    bool is_empty() {
        return items.empty();
    }
};

int main() {
    myQueue q;
    q.enqueue("a");
    q.enqueue("b");
    q.enqueue("c");

    cout << q.dequeue() << endl;  // a
    cout << q.dequeue() << endl;  // b
    cout << q.dequeue() << endl;  // c

    return 0;
}
